src/sql/workbench/parts/notebook/notebook.component.ts

```diff
@@ -523,11 +523,20 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 		}
 	}
 
-	public async runAllCells(): Promise<boolean> {
+	public async runAllCells(startId?: string, endId?: string): Promise<boolean> {
 		await this.modelReady;
 		let codeCells = this._model.cells.filter(cell => cell.cellType === CellTypes.Code);
 		if (codeCells && codeCells.length) {
-			for (let i = 0; i < codeCells.length; i++) {
+			// For the run all cells scenario where neither startId not endId are provided, set defaults
+			let startIndex = 0;
+			let endIndex = codeCells.length;
+			if (startId) {
```

Comment:
What if it's 0? Shouldn't we use 
```ts
if (!types.isUndefinedOrNull(startId) {
```

Reply:
Didn't even know that this method existed. Love it.